.github/workflows/testing.yml

```diff
@@ -89,63 +72,82 @@ jobs:
         run: |
           sudo apt install libmagic1
 
-      # Install and test - Test order is randomized, run three times to ensure correctness
-      - name: Reset local database
+      - name: Create _db/secrets.py
         run: |
           tar -zxvf _db.initial.tar.gz
           mv _db.initial _db
-          ./scripts/install.sh
 
       - name: Create frontend static directory (tests complains if static directory does not exist)
         run: |
           mkdir -p app/static/dist-latest
           touch app/static/dist-latest/404.html
           touch app/static/dist-latest/index.html
 
+      - name: Run docker-compose
+        run: |
+          set -ex
+          cd deploy/codex
+          chmod a+r .env
+          docker-compose up -d db redis gitlab edm houston
+          # Check the state of the containers
+          sleep 1m
+          # Wait until houston is up
+          while sleep 15
+          do
+            docker-compose logs houston | tail
+            docker-compose ps
+            if [ -n "$(docker-compose ps | grep Exit)" ]
+            then
+              exit 1
+            fi
+            wget --tries=1 -O - http://localhost:83/ && break
+          done
+        env:
+          # docker-compose.yml mounts ../.. so we need to be in the
+          # deploy/codex directory
+          COMPOSE_FILE: 'docker-compose.yml'
+
+      # Install and test - Test order is randomized, run three times to ensure correctness
       - name: Run tests (random x3)
         run: |
-          source virtualenv/houston3.7/bin/activate
-          pytest -s -v --gitlab-remote-login-pat "${{ secrets.GITLAB_REMOTE_LOGIN_PAT }}"
-          pytest -s -v --gitlab-remote-login-pat "${{ secrets.GITLAB_REMOTE_LOGIN_PAT }}"
-          pytest -s -v --gitlab-remote-login-pat "${{ secrets.GITLAB_REMOTE_LOGIN_PAT }}"
+          docker-compose exec -T -e FLASK_CONFIG=$FLASK_CONFIG -e SQLALCHEMY_DATABASE_URI=$SQLALCHEMY_DATABASE_URI houston pytest -s -v
```

Comment:
Just wondering if there was any way to make 
docker-compose exec -T -e FLASK_CONFIG=$FLASK_CONFIG -e SQLALCHEMY_DATABASE_URI=$SQLALCHEMY_DATABASE_URI houston
an alias of some kind in yml.
The text is now used multiple times in this file and it's not immediately obvious that they are all the same (I think they are)

Reply:
I'm not entirely sure what you're talking about...  `FLASK_CONFIG`  isn't defined in `testing.yml` and so it should be `''` and I needed to pass it in because otherwise it's using the default `FLASK_CONFIG` in the docker container.  `SQLALCHEMY_DATABASE_URI` is also defined in the docker container (with postgres db uri) so I needed to override it when running sqlite tests.

Reply:
I was just thinking that the text block "docker-compose exec -T -e FLASK_CONFIG=$FLASK_CONFIG -e SQLALCHEMY_DATABASE_URI=$SQLALCHEMY_DATABASE_URI houston" was duplicated quite a few times. 
If this was something like a shell script, it would be clearer if was some sort of alias. not sure how this could be done in yml

Reply:
The inclusion  of the `FLASK_CONFIG` env-var in this invocation is because our tests don't play friendly when the env-var is set. So this is more of a work around to that issue.
This issue seemed pretty low on the todo list for me, but we likely shouldn't continue to work around it. Yinz aren't actively using this env-var on a daily basis or the composition for testing AFAIK, so it's been low priority.